shimming-toolbox/shimmingtoolbox/load_nifti.py

```diff
@@ -12,34 +12,167 @@
 
 logger = logging.getLogger(__name__)
 PHASE_SCALING_SIEMENS = 4096
+POSSIBLE_TIMINGS = ['slice-middle', 'volume-start', 'volume-middle']
 
 
-def get_acquisition_times(nii_data, json_data):
-    """Return the acquisition timestamps from a json sidecar. This assumes BIDS convention.
+def get_acquisition_times(nii_data, json_data, when='slice-middle'):
+    f""" Return the acquisition timestamps from a json sidecar. This assumes BIDS convention.
 
     Args:
         nii_data (nibabel.Nifti1Image): Nibabel object containing the image timeseries.
         json_data (dict): Json dict corresponding to a nifti sidecar.
+        when (str): When to get the acquisition time. Can be within {POSSIBLE_TIMINGS}.
 
     Returns:
-        numpy.ndarray: Acquisition timestamps in ms.
-
+        numpy.ndarray: Acquisition timestamps in ms (n_volumes x n_slices).
     """
+
+    if when not in POSSIBLE_TIMINGS:
+        raise ValueError(f"Invalid 'when' parameter. Must be within {POSSIBLE_TIMINGS}")
+
     # Get number of volumes
     n_volumes = nii_data.header['dim'][4]
+    n_slices = nii_data.shape[2]
+
+    # Time between the beginning of the acquisition of a volume and the beginning of the acquisition of the next volume
+    deltat_volume = float(json_data['RepetitionTime']) * 1000  # [ms]
+
+    # Time the acquisition of data for this image started (ISO format)
+    acq_start_time_iso = json_data['AcquisitionTime']
+    # todo: dummy scans?
```

Comment:
I don't think dummy scan affects the start time, but it needs to be tested somehow